ref: extract technical details from `exp run` to guides (#3182)
iterative/dvc.org: content/docs/command-reference/exp/run.md

can execute them one by one (default) or in parallel (using the `--jobs`
option).

Instead of a single commit, checkpoint experiments have multiple commits under
the custom Git reference (in `.git/refs/exps`), similar to a branch.
> 📖 Learn more about the [experiments queue].

</details>

List previous checkpoints with `dvc exp show`. To resume from a previous
checkpoint, you must first `dvc exp apply` it before using `dvc exp run`. For
`--queue` or `--temp` runs (see next section), use `--rev` instead to specify
the checkpoint to continue from.

Alternatively, use `--reset` to start over (discards previous checkpoints and
their outputs). This is useful for re-training ML models, for example.

## Queueing and parallel execution

The `--queue` option lets you create an experiment as usual, except that nothing
is actually run. Instead, the experiment is put in a wait-list for later
execution. `dvc exp show` will mark queued experiments with an asterisk `*`.
It's also possible to run special [checkpoint experiments] for deep learning ML.

> Note that queuing an experiment that uses checkpoints implies `--reset`,

> > unless a `--rev` is provided (refer to the previous section).

Author: Q: I removed this note from the ref. and it's not mentioned in any guide (it's still noted in the Options section for now). Should it be mentioned in a guide? Where? How?

<abbr>project</abbr> without polluting it with unnecessary commits, branches,
directories, etc.

> `dvc exp run` is equivalent to `dvc repro` for experiments. It has the same
> behavior when it comes to `targets` and stage execution (restores the
> dependency graph, etc.). See the command [options](#options) for more on the
> differences.
> 📖 See full [Running Experiments] guide for more information.

Before running an experiment, you'll probably want to make modifications such as
data and code updates, or <abbr>hyperparameter</abbr> tuning. For the latter,
you can use the `--set-param` (`-S`) option of this command to change
`dvc param` values on-the fly.
<abbr>parameter</abbr> tuning. You can use the `--set-param` (`-S`) option to
change param values on-the fly.

Each experiment creates and tracks a project variation based on your
<abbr>workspace</abbr> changes. Experiments will have a unique, auto-generated
name like `exp-bfe64` by default, which can be customized using the `--name`

> (`-n`) option.

Author: UPDATE: Per https://github.com/iterative/dvc.org/pull/3182#issuecomment-1012558741 I reintroduced text that describes all major features of `exp run` to this ref.

## Example: Modify parameters on-the-fly

`dvc exp run--set-param` (`-S`) saves you the need to manually edit the params
file before running an experiment.

```dvc
$ dvc exp run -S prepare.split=0.25 -S featurize.max_features=2000
...
Reproduced experiment(s): exp-18bf6
```

> To see the results, we can use `dvc exp diff`, which compares both params and

Reviewer: I think the sentence could be split into two. 

We can use `dvc exp diff` to see the result. It compares ...

Author: Sure. Also it should use 2nd person now that I see it:
```suggestion
To see the results, you can use `dvc exp diff`. It compares both params and
```
